lib/solana.cpp

```diff
@@ -495,6 +495,20 @@ Connection::getSignatureStatus(const std::string &signature,
   return {res.context, res.value[0]};
 }
 
+bool Connection::confirmTransaction(std::string transactionSignature,
```

Comment:
You can remove the timeout part. It is the client who will implement the timeout and retry logic. You can just implement confirmTransaction part, usually we avoid sleep except for tests.

Reply:
Ahh I will remove that

Reply:
@godmodegalactus i disagree, Confirm transaction is not a low level wrapper around get signature status but a high level api that has complex control flow. Loops, multiple threads and so on, all packaged up as a convenience method that you can throw in your code to make sure the transaction got confirmed with a configurable level

Reply:
I think it makes more sense to tackle this one though once the websocket module is ready and would hold off for now on any changes. Let @pSN0W have a look at this when he has a basic websocket working. ill open a new issue for this method alone and assign it to him.

Reply:
If you want to make an intermediate version it should use:
a) blockhash expiry (not timeouts)
b) a loop updating block heigh and signature status

Reply:
> I think it makes more sense to tackle this one though once the websocket module is ready and would hold off for now on any changes. Let @pSN0W have a look at this when he has a basic websocket working. ill open a new issue for this method alone and assign it to him.

Sorry for the confusion: please go ahead with an intermediate version (blockheight not timeout), to finish up the PR, and then reassign issue to @pSN0W so he can finish it once the websocket part is ready

Reply:
@mschneider Yes I agree with you. With websockets implemented we can implement a high level confrim transaction. I was more commenting about sleeping till confirmation is not the correct way. But if it is ok for intermediate step then its a go for me.

Reply:
So, if I am not very wrong a possible way to implement this currently would to to have 
like a call to getBlockHeight and check whenever the blockHeight increases if the getSignature status returns a "finalized" or "confirmed"?

Reply:
The timeout should not be uint8_t but something like timepoint or duration etc. It also should be optional.

Reply:
https://github.com/solana-labs/solana-web3.js/blob/eaafcf0/src/connection.ts#L3334 check this method for reference

Reply:
https://github.com/blockworks-foundation/mango-client-v3/blob/0ad1e0637dca574e5140cb1e86c14fa2be70be7f/src/client.ts#L546 this is what mango uses, and it shows very well how it can work without websockets, it also shows how the different confirmation levels are related

Reply:
already realized in  one of examples. yo have blockheight part too somewhere there.
https://github.com/alexeyneu/solcpp/blob/bc6bf842bfb85f274527163496042b0430cf3f82/examples/placeOrder.cpp#L212-L253
  
```c++
      const auto timeoutBlockHeight =
        recentBlockHash.lastValidBlockHeight +
        mango_v3::MAXIMUM_NUMBER_OF_BLOCKS_FOR_TRANSACTION;
      do{
      while (!false) {
        // Check if we are past validBlockHeight
        auto currentBlockHeight = connection.getBlockHeight("confirmed");
        if (timeoutBlockHeight <= currentBlockHeight) {
          spdlog::error("Timed out for txid: {}, current BlockHeight: {}", b58Sig,
                        currentBlockHeight);
          break;
        }
        const json req = connection.getSignatureStatuses({b58Sig});
        const std::string jsonSerialized = req.dump();
        spdlog::info("REQ: {}", jsonSerialized);
	std::stringstream ent_f;
	struct curl_slist *headers = NULL;
	headers = curl_slist_append(headers, "Content-Type: application/json");

	curl_easy_setopt(f, CURLOPT_URL, config.endpoint.c_str());
	curl_easy_setopt(f, CURLOPT_COPYPOSTFIELDS, jsonSerialized.c_str());
	curl_easy_setopt(f, CURLOPT_POST, 1);
	curl_easy_setopt(f, CURLOPT_WRITEFUNCTION, CurlWrite);
    	curl_easy_setopt(f, CURLOPT_WRITEDATA, &ent_f);
	curl_easy_setopt(f, CURLOPT_FAILONERROR, 1);
	curl_easy_setopt(f, CURLOPT_HTTPHEADER, headers);
	CURLcode b = curl_easy_perform(f);

	curl_slist_free_all(headers);
        if (b == CURLE_OK) {
                std::string e_f(ent_f.str());
		spdlog::info("RES: {}", e_f.c_str());
		json res = json::parse(e_f.c_str());
	        if (res["result"]["value"][0] != nullptr) { sell.first.second= false;buy.second= false;sell.second = !false; break; }

        } else {
		spdlog::error("Error: {}",   curl_easy_strerror(b));
		break;
	}

      }
      }while(buy.second == !false);
```

Reply:
Ahh I will make these changes.

Reply:
I think it's done